pythonFiles/testing_tools/adapter/util.py

```diff
@@ -170,17 +170,36 @@ def fix_fileid(
 # stdio
 
 
+@contextlib.contextmanager
+def _hide_buffer(attr, ignored):
+    # an alternative approach is suggested: #6594
+    assert attr in ("stdout", "stderr")
+    stdout_fd = getattr(sys, attr).fileno()
+    # Set / unset sys.stdout or sys.stderr values.
+    # This will swallow python print statements, logs, etc.
+    setattr(sys, attr, ignored)
+    try:
+        with os.fdopen(os.dup(stdout_fd), "w") as old_stdout:
+            # Set / unset file descriptors.
+            # This is required for output coming out of layers lower than python,
+            # e.g., C printf statements via cython.
+            with open(os.devnull, "w") as devnull:
+                os.dup2(devnull.fileno(), stdout_fd)
+            try:
+                yield ignored
+            finally:
+                os.dup2(old_stdout.fileno(), stdout_fd)
```

Comment:
As it is written, this can confuse readers that don't know how file descriptors work.  So let's make it more clear:

```suggestion
            with open(os.devnull, "w") as devnull:
                os.dup2(devnull.fileno(), stdout_fd)
                try:
                    yield ignored
                finally:
                    os.dup2(old_stdout.fileno(), stdout_fd)
```